app/auth/TokenSecurity.scala

```diff
@@ -4,21 +4,38 @@ import com.nimbusds.oauth2.sdk.token.BearerAccessToken
 import configuration.KeycloakConfiguration
 import org.pac4j.core.profile.{CommonProfile, ProfileManager}
 import org.pac4j.play.PlayWebContext
-import play.api.mvc.{AnyContent, Request}
-import uk.gov.nationalarchives.tdr.keycloak.{KeycloakUtils, TdrKeycloakDeployment, Token}
+import play.api.i18n.I18nSupport
+import play.api.mvc.{Action, AnyContent, Request, Result}
+import uk.gov.nationalarchives.tdr.keycloak.Token
 
-import scala.concurrent.ExecutionContext
+import scala.concurrent.Future
 
-trait TokenSecurity extends OidcSecurity {
+trait TokenSecurity extends OidcSecurity with I18nSupport {
 
   def keycloakConfiguration: KeycloakConfiguration
 
-  implicit def requestToRequestWithToken(request: Request[AnyContent])(implicit executionContext: ExecutionContext): RequestWithToken = {
+  implicit def requestToRequestWithToken(request: Request[AnyContent]): RequestWithToken = {
     val webContext = new PlayWebContext(request, playSessionStore)
     val profileManager = new ProfileManager[CommonProfile](webContext)
     val profile = profileManager.get(true)
     val token: BearerAccessToken = profile.get().getAttribute("access_token").asInstanceOf[BearerAccessToken]
     val accessToken: Option[Token] = keycloakConfiguration.token(token.getValue)
     RequestWithToken(request, accessToken)
   }
+
+  def judgmentUserAction(block: Request[AnyContent] => Future[Result]): Action[AnyContent] = secureAction.async { request =>
```

Comment:
Ah yeah, I think the code I originally copying this from was blocking and I didn't change the name. I've renamed it to action because it does take an action as a parameter.